### ISSUE TITLE
scan+pmap shape bug

### ISSUE THREAD

**Issue by mattjj** [google/jax#1369]
@JasperSnoek reported this one:

```python
import jax.numpy as np
from jax import lax, random
from jax import pmap

from jax.lib import xla_bridge
nrep = xla_bridge.device_count()

key = random.PRNGKey(0)

n = 80
a_in = np.linspace(0, 1, n)[:, None]
a = np.dot(a_in, a_in.T) + np.eye(n) * 0.1
b = random.normal(key, (n,)).flatten()

def pmvm(a, b):
  a = a.reshape((nrep, -1, a.shape[1]))
  func = pmap(lambda z: np.dot(z, b))
  return func(a).reshape(b.shape)

for i in range(10):
  b = pmvm(a, b)
# Ok

iters = np.arange(10)
def body(carry, i):
  return pmvm(a, carry), i
res = lax.scan(body, b, iters)
```

It results in some kind of shape error (the exact one depends on how many replicas there are).

**Comment by rkruegs123** [comment]
@mattjj This is a great example, thank you!

Suppose you were to take the gradient of `res` (or a scalar that is a function of `res`). How would the gradient calculation be distributed across devices per the `pmap` call?

I also asked on the Discussion board here: https://github.com/google/jax/discussions/19393

**Comment by rkruegs123** [comment]
Hi @mattjj , just seeing if you have any thoughts on this. Thank you in advance.
